### PR TITLE
New schema locater to locate the standard schemas delivered as part of the core-backend package

### PR DESCRIPTION
**Pull request by RohitPtnkr1996**
Fixes https://github.com/iTwin/itwinjs-core/issues/6145

### CODE REVIEW

File: core/ecschema-locaters/src/SchemaJsonFileLocater.ts [hunk 1]
```diff
@@ -116,3 +116,80 @@ export class SchemaJsonFileLocater extends SchemaFileLocater implements ISchemaL
...
   }
 }
+
+/**
+ * A SchemaLocator implementation for locating JSON Schema files
+ * from the file system using configurable search paths.
+ * @beta This is a workaround the current lack of a full xml parser.
+ */
+export class PublishedSchemaJsonFileLocater extends SchemaJsonFileLocater implements ISchemaLocater {
+  public static defaultSchemaSearchPaths = new Set([
```
Comment (aruniverse) [inline core/ecschema-locaters/src/SchemaJsonFileLocater.ts:126]: this seems to be repeated a few times, wonder if we extract this out to a var that is referenced everywhere
what happens today if you update one of these places but not the others?

having a list of these schemas grouped is helpful, but if its not needed you could use a glob to pull in all schemas

Comment (RohitPtnkr1996) [reply core/ecschema-locaters/src/SchemaJsonFileLocater.ts]: Done!

**Comment by ColinKerr** [comment]
I don't think this is the right solution to the problem.

This change adds three locaters.  Only one is necessary.  A schema context takes multiple locaters so if a user wants to add paths they can add additional locaters to the context.  

This change also combines ability to load from packages installed in node_modules with the requested ability to load schemas deployed with the backend.  Loading assuming node_modules is fixed has limited use as things may be packaged in a real deployment.

I suggest implementing one locater that takes the backend assets dir string (must be defined) and internally adds paths for the sub directories needed.  It should not accept any other paths from the user.  It should be explicitly documented so the user knows how to construct it and what schemas it will locate.

Locaters added to a context are checked in the order they are added, so the last locater added is the last one checked.  So it is possible to construct a schema context with this locater as the lowest precedence locater but that is easy to mess up.  One possible strategy to improve this would be to follow the pattern from imodel_native where you can add a last chance locater.

It would be better if it 'just worked' without having to take a path or requiring the user to register this locater correctly but that is not possible because core-backend does not have a dependency on any of the ecschema packages.

**Comment by RohitPtnkr1996** [comment]
Okay sounds good ! I will strip down the locater to focus only on the backend schemas.
I was thinking of this as a possible replacement/extension of the existing locater with the added functionality of being able to locater backend schemas.
I added the node-modules search code to be able to load in a wider set of schemas including their json versions as the backend only puts out xml files.

File: core/ecschema-locaters/src/SchemaXmlFileLocater.ts [hunk 2]
```diff
@@ -120,3 +122,43 @@ export class SchemaXmlFileLocater extends SchemaFileLocater implements ISchemaLo
...
+    this.searchPaths.push(...globSync(path.join(knownBackendAssetsDir, "ECSchemas", "*/"), { windowsPathsNoEscape: true }).filter(fs.existsSync));
+  }
+
+  /**
+   * Overrides the addSchemaSearchPath method to prevent adding new schema search paths.
+   *
+   * @param _schemaPath - The schema path to add (ignored).
+   */
+  public override addSchemaSearchPath(_schemaPath: string): void {
+    return; // Do nothing, this is a read-only locator
```
Comment (rschili) [inline core/ecschema-locaters/src/SchemaXmlFileLocater.ts:153]: Can you elaborate on this? I'd rather actually add to searchPaths, or throw here.

**Review by rschili — APPROVED** [review]

File: core/ecschema-locaters/src/SchemaXmlFileLocater.ts (hunk 2 above)
Comment (rschili) [reply core/ecschema-locaters/src/SchemaXmlFileLocater.ts:157]: Okay, Rohit explained it to me. It still feels odd, but I understand the reasoning. Closing the conversation.

**Review by StefanApfel-Bentley — APPROVED** [review]

**Pull request merged**